### ISSUE TITLE
--after-upgrade with existing --after-install makes the latter execute before install (rpm)

### ISSUE THREAD

**Issue by Rush** [jordansissel/fpm#830]
That's how we generate our package, all is well with this!
```bash
function generate_rpm {
    fpm --workdir `pwd`/tmp --rpm-compression=xz -s dir -t rpm \
    "$@" -n "virtkick" -v "$VERSION" \
    --iteration "$ITERATION" \
    -p "virtkick-VERSION-ITERATION${SUFFIX}_ARCH.rpm" \
    -m "Damian Kaczmarek <rush@virtkick.io>" \
    --after-install scripts/rpm/after_install.sh \
    --before-remove scripts/rpm/before_remove.sh \
    -C root_package \
    opt usr
}
```
Unfortunately adding an `--after-upgrade` completely breaks the above setup and generated package suddenly starts executing post install scripts as pre install scripts.
```bash
function generate_rpm {
    fpm --workdir `pwd`/tmp --rpm-compression=xz -s dir -t rpm \
    "$@" -n "virtkick" -v "$VERSION" \
    --iteration "$ITERATION" \
    -p "virtkick-VERSION-ITERATION${SUFFIX}_ARCH.rpm" \
    -m "Damian Kaczmarek <rush@virtkick.io>" \
    --after-upgrade scripts/rpm/after_upgrade.sh \
    --after-install scripts/rpm/after_install.sh \
    --before-remove scripts/rpm/before_remove.sh \
    -C root_package \
    opt usr
}
```

The contents of RPM speak for themselves. Before adding `--after-upgrade` contents are good, `POSTIN` shows proper content. After adding `--after-upgrade` `POSTIN` is non-existent and `INFO/SCRIPTS/PREIN` shows:
```rpm
upgrade() {
# Rush: contents of before-install script are being moved to here
}
install() {
# Rush: contents of after-install script are being moved to here
}
if [ "${1}" -eq 1 ]
then
    install
    # "before install" goes here <---- THIS IS YOUR COMMENT
elif [ "${1}" -gt 1 ]
then
    upgrade
fi
```
The above source basically shows up the bugs by itself. After install as treated as before install.
Adding @Nowaker as interested party.

**Comment by Nowaker** [comment]
@jordansissel Any chance to get this fixed? Seems like fixed in https://github.com/xyntrix/fpm/commit/6553e66f3842d4726515099800588bd87a00608b.

**Comment by ssenaria** [comment]
Looks like we got hit by this bug as well. `--after-upgrade` doesn't do anything for us. It actually removes the symlinks!

**Comment by mateuszkwiatkowski** [comment]
Hello,
Any chance to fix it?

**Comment by paszczak000** [comment]
I have the same issue. Adding `--before-upgrade` or `--after-upgrade` is breaking buidling of RPMs.

````
Process failed: rpmbuild failed (exit code 1). Full command was:["rpmbuild", "-bb", "--define", "buildroot /tmp/package-rpm-build20150708-6277-1fg029u/BUILD", "--define", "_topdir /tmp/package-rpm-build20150708-6277-1fg029u", "--define", "_sourcedir /tmp/package-rpm-build20150708-6277-1fg029u", "--define", "_rpmdir /tmp/package-rpm-build20150708-6277-1fg029u/RPMS", "--define", "_tmppath /tmp", "/tmp/package-rpm-build20150708-6277-1fg029u/SPECS/tc-zabbix.spec"] {:level=>:error}
````

**Comment by Nowaker** [comment]
@paszczak000 This issue isn't about getting any error messages. Please report a new issue.

**Comment by robertwatson3** [comment]
--after-upgrade is breaking my install as well. I'm invoking fpm with "--after-upgrade upgrade.sh", but for some reason, the contents of upgrade.sh are being used as the preinstall scriptlet:
```
[vagrant@localhost ~]$ rpm -qp [rpm created with fpm] --scripts
preinstall scriptlet (using /bin/sh):
upgrade() {
#/bin/sh
[upgrade script]
}
install() {
#/bin/sh
[install script]
}
if [ "${1}" -eq 1 ]
then
    install
    # "before install" goes here
elif [ "${1}" -gt 1 ]
then
    upgrade
fi
preuninstall scriptlet (using /bin/sh):
if [ "${1}" -eq 0 ]
then
[uninstall script]

fi
```

**Comment by robertwatson3** [comment]
if I run fpm with the "-e" option, and just change "%pre" to "%post", it works perfectly. Unfortunately, we're using an automated build system, so manually editing the package spec is not an option.

**Comment by robertwatson3** [comment]
This is actually fixed in the code that is hosted on github, it is not fixed in the code that is hosted on ruby gems. At the time this comment was posted, the gem hadn't been updated since December 11, 2014. So if you're still having an issue with this, use this command to update fpm:
```
  sudo gem install specific_install
  sudo gem specific_install https://github.com/jordansissel/fpm.git
```

**Comment by paszczak000** [comment]
@robertwatson3 specific_install for gem fix problem. Thanks!

**Comment by Rush** [comment]
@paszczak000 well the issue is sitll open in the context of this repo, please keep it open so that other may easily find the solution.

**Comment by robertwatson3** [comment]
The only problem with this fix is that both versions (the rubygems version that hasn't been updated since 2014, and the up to date version on github) are still at 1.3.3. So we can't check to see if the version installed actually fixes this issue or not in our build environment. Instead, we have to force the specific_install.

Can we get a 1.3.4 pushed to ruby gems?

**Comment by Rush** [comment]
This is fixed by fpm version 1.4.0 which has been release last weekend, thank you!
